src/internal/connector/graph/service.go

```diff
@@ -36,6 +36,7 @@ const (
 	rateLimitHeader           = "RateLimit-Limit"
 	rateRemainingHeader       = "RateLimit-Remaining"
 	rateResetHeader           = "RateLimit-Reset"
+	defaultHTTPClientTimeout  = 1 * time.Hour
```

Comment:
1 hour feels a bit too long IMO, any reason why we have such a big timeout? I doubt any request would take >10-15min.

Reply:
Ahh, I see this change is already in and this is just restructuring the code.